src/aws/flb_aws_credentials_iot.c

```diff
@@ -0,0 +1,655 @@
+                                        NULL, /* ca_path */
+                                        implementation->ca_cert_file,
+                                        implementation->cert_file,
+                                        implementation->key_file,
+                                        NULL); /* key_passwd */
+    if (!implementation->tls) {
+        flb_error("[aws_credentials] Failed to create TLS instance for IoT Provider");
+        goto error;
+    }
+    
+    flb_debug("[aws_credentials] TLS instance created successfully");
+
+    /* Create upstream connection */
+    flb_debug("[aws_credentials] Creating upstream connection to %s:%d", host, port);
+    upstream = flb_upstream_create(config, host, port, FLB_IO_TLS, implementation->tls);
+    if (!upstream) {
+        flb_error("[aws_credentials] IoT Provider: connection initialization error");
+        goto error;
+    }
+    
+    flb_debug("[aws_credentials] Upstream connection created successfully");
+
+    upstream->base.net.connect_timeout = FLB_AWS_CREDENTIAL_NET_TIMEOUT;
+
+    implementation->client = generator->create();
+    if (!implementation->client) {
+        flb_aws_provider_destroy(provider);
+        flb_upstream_destroy(upstream);
+        flb_error("[aws_credentials] IoT Provider: client creation error");
+        return NULL;
+    }
+
+    implementation->client->name = "iot_provider_client";
+    implementation->client->has_auth = FLB_FALSE;
+    implementation->client->provider = NULL;
+    implementation->client->region = NULL;
+    implementation->client->service = NULL;
+    implementation->client->port = port;
+    implementation->client->flags = 0;
+    implementation->client->proxy = NULL;
+    implementation->client->upstream = upstream;
+    
+    flb_debug("[aws_credentials] IoT client configured: name=%s, port=%d, has_auth=%d", 
+              implementation->client->name, implementation->client->port, implementation->client->has_auth);
+
+    /* Set up the thing name header */
+    implementation->thing_name_header.key = "x-amzn-iot-thingname";
+    implementation->thing_name_header.key_len = 22;
+    implementation->thing_name_header.val = implementation->thing_name;
+    implementation->thing_name_header.val_len = strlen(implementation->thing_name);
+
+    flb_debug("[aws_credentials] Setting IoT thing name header: %s = %s", 
+              implementation->thing_name_header.key, implementation->thing_name_header.val);
+
+    /* Set the static headers for the client */
+    implementation->client->static_headers = &implementation->thing_name_header;
+    implementation->client->static_headers_len = 1;
+
+    cleanup:
+        flb_sds_destroy(protocol);
+        flb_sds_destroy(host);
+        flb_sds_destroy(port_sds);
+        flb_sds_destroy(endpoint_path);
+        return provider;
+    error:
+        flb_aws_provider_destroy(provider);
+        provider = NULL;
+        goto cleanup;
```

Comment:
Not critical but for the protocol
```suggestion
    error:
        flb_aws_provider_destroy(provider);
        provider = NULL;
    cleanup:
        flb_sds_destroy(protocol);
        flb_sds_destroy(host);
        flb_sds_destroy(port_sds);
        flb_sds_destroy(endpoint_path);
        return provider;
```